<?xml version="1.0" encoding="UTF-8" ?>
<rss version="2.0" xmlns:atom="http://www.w3.org/2005/Atom" xmlns:dc="http://purl.org/dc/elements/1.1/">
  <channel>
    <title>Caught exception!</title>
    <link>http://community.axoloti.com/t/caught-exception/400</link>
    <description>.. And I&#39;m feeling proud as a cat with a mouse! ;)

This happened when triggering samples with a Korg Nanopad 2.

Control transfer failed: -7
connected
java.lang.Exception: cpuserial has wrong length
	at axoloti.HWSignature.Verify(HWSignature.java:96)
	at axoloti.USBBulkConnection.connect(USBBulkConnection.java:321)
	at axoloti.MainFrame.jCheckBoxConnectActionPerformed(MainFrame.java:769)
	at axoloti.MainFrame.access$400(MainFrame.java:80)
	at axoloti.MainFrame$7.actionPerformed(MainFrame.java:410)
	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)
	at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2346)
	at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)
	at javax.swing.JToggleButton$ToggleButtonModel.setPressed(JToggleButton.java:308)
	at javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:252)
	at java.awt.Component.processMouseEvent(Component.java:6525)
	at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)
	at java.awt.Component.processEvent(Component.java:6290)
	at java.awt.Container.processEvent(Container.java:2234)
	at java.awt.Component.dispatchEventImpl(Component.java:4881)
	at java.awt.Container.dispatchEventImpl(Container.java:2292)
	at java.awt.Component.dispatchEvent(Component.java:4703)
	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4898)
	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4533)
	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4462)
	at java.awt.Container.dispatchEventImpl(Container.java:2278)
	at java.awt.Window.dispatchEventImpl(Window.java:2750)
	at java.awt.Component.dispatchEvent(Component.java:4703)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:86)
	at java.awt.EventQueue$4.run(EventQueue.java:731)
	at java.awt.EventQueue$4.run(EventQueue.java:729)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)</description>
    
    <lastBuildDate>Sat, 05 Sep 2015 13:19:44 +0000</lastBuildDate>
    <category>Helpdesk</category>
    <atom:link href="https://sebiik.github.io/community.axoloti.com.backup/t/caught-exception/400.rss" rel="self" type="application/rss+xml" />
      <item>
        <title>Caught exception!</title>
        <dc:creator><![CDATA[@mackemint Markus Kaczmarek]]></dc:creator>
        <description><![CDATA[
          <p><a href="https://sebiik.github.io/community.axoloti.com.backup/u/mackemint">@mackemint</a> wrote:</p>
          <blockquote>
              <p>Good, thanks for the info even tho it is as you say, a tad unspecific! :3</p>

<p>How bout crashes when not running in live mode, will the console return error messages anyway?</p>
          </blockquote>
          <p><a href="https://sebiik.github.io/community.axoloti.com.backup/t/caught-exception/400/6">Read full topic</a></p>
        ]]></description>
        <link>http://community.axoloti.com/t/caught-exception/400/6</link>
        <pubDate>Sat, 05 Sep 2015 13:19:44 +0000</pubDate>
        <guid isPermaLink="false">community.axoloti.com-post-400-6</guid>
        <source url="http://community.axoloti.com/t/caught-exception/400.rss">Caught exception!</source>
      </item>
      <item>
        <title>Caught exception!</title>
        <dc:creator><![CDATA[@thetechnobear Mark Harris]]></dc:creator>
        <description><![CDATA[
          <p><a href="https://sebiik.github.io/community.axoloti.com.backup/u/thetechnobear">@thetechnobear</a> wrote:</p>
          <blockquote>
              <p>ok, i checked the source code... they are basically errors writing to USB </p>

<p>Ive details them in a new topic (and probably will expand later), but as mentioned they don't really tell us much more than we could not write to the board  ... -7 is Timeout, this probablymeans board is not 'acking' the data sent to it, probably because its crashed, so not processing USB packets.</p>

<p><a href="/t/axoloti-error-codes">Axoloti error codes</a></p>
          </blockquote>
          <p><a href="https://sebiik.github.io/community.axoloti.com.backup/t/caught-exception/400/5">Read full topic</a></p>
        ]]></description>
        <link>http://community.axoloti.com/t/caught-exception/400/5</link>
        <pubDate>Sat, 05 Sep 2015 11:22:41 +0000</pubDate>
        <guid isPermaLink="false">community.axoloti.com-post-400-5</guid>
        <source url="http://community.axoloti.com/t/caught-exception/400.rss">Caught exception!</source>
      </item>
      <item>
        <title>Caught exception!</title>
        <dc:creator><![CDATA[@thetechnobear Mark Harris]]></dc:creator>
        <description><![CDATA[
          <p><a href="https://sebiik.github.io/community.axoloti.com.backup/u/thetechnobear">@thetechnobear</a> wrote:</p>
          <blockquote>
              <aside class="quote" data-post="3" data-topic="400"><div class="title">
<div class="quote-controls"></div>
<img alt="" width="20" height="20" src="//community.axoloti.com/user_avatar/community.axoloti.com/mackemint/40/27_1.png" class="avatar">mackemint:</div>
<blockquote><p>Where can I read up on the failure codes?</p></blockquote></aside>

<p>that is a very good question... <a class="mention" href="/users/johannes">@johannes</a> ?</p>

<p>I've so far seen -1,-4, -7, -9  ...  If Johannes doesn't know off hand, I'll go check the code.<br>and then put something in the user guide section (I suspect the codes won't tell us a huge amount)</p>
          </blockquote>
          <p><a href="https://sebiik.github.io/community.axoloti.com.backup/t/caught-exception/400/4">Read full topic</a></p>
        ]]></description>
        <link>http://community.axoloti.com/t/caught-exception/400/4</link>
        <pubDate>Sat, 05 Sep 2015 11:10:31 +0000</pubDate>
        <guid isPermaLink="false">community.axoloti.com-post-400-4</guid>
        <source url="http://community.axoloti.com/t/caught-exception/400.rss">Caught exception!</source>
      </item>
      <item>
        <title>Caught exception!</title>
        <dc:creator><![CDATA[@mackemint Markus Kaczmarek]]></dc:creator>
        <description><![CDATA[
          <p><a href="https://sebiik.github.io/community.axoloti.com.backup/u/mackemint">@mackemint</a> wrote:</p>
          <blockquote>
              <p>Yeah, this is what happens when you're trying to debug in the middle of the night when u should have gone to sleep hours ago..<br>Probably won't be able to reproduce it, but I'll get back 2 u if I do.</p>

<p>Where can I read up on the failure codes?</p>
          </blockquote>
          <p><a href="https://sebiik.github.io/community.axoloti.com.backup/t/caught-exception/400/3">Read full topic</a></p>
        ]]></description>
        <link>http://community.axoloti.com/t/caught-exception/400/3</link>
        <pubDate>Sat, 05 Sep 2015 10:57:56 +0000</pubDate>
        <guid isPermaLink="false">community.axoloti.com-post-400-3</guid>
        <source url="http://community.axoloti.com/t/caught-exception/400.rss">Caught exception!</source>
      </item>
      <item>
        <title>Caught exception!</title>
        <dc:creator><![CDATA[@thetechnobear Mark Harris]]></dc:creator>
        <description><![CDATA[
          <p><a href="https://sebiik.github.io/community.axoloti.com.backup/u/thetechnobear">@thetechnobear</a> wrote:</p>
          <blockquote>
              <aside class="quote" data-post="1" data-topic="400"><div class="title">
<div class="quote-controls"></div>
<img alt="" width="20" height="20" src="//community.axoloti.com/user_avatar/community.axoloti.com/mackemint/40/27_1.png" class="avatar">mackemint:</div>
<blockquote><p>at javax.swing.JToggleButton$ToggleButtonModel.setPressed(JToggleButton.java:308)	at javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:252)	at java.awt.Component.processMouseEvent(Component.java:6525)</p></blockquote></aside>

<p>are you sure your cat didnt tread on the mouse?  <img src="/images/emoji/emoji_one/smile.png?v=0" title=":smile:" class="emoji" alt="smile">  (my kittens do all the time!)</p>

<p>seriously though...  to me this looks like, you had data failure (-7)  , and then you pressed 'connect' to reconnect.<br>and this failed, to verify the serial number..</p>

<p>assuming this is correct... can you tell me ...</p>

<ul>
<li>did the patch continue to run, even though not connected (patches don't need the UI to run, so sometimes the UI gets disconnected but the patch carries on fine)</li>
<li>did you have to power cycle your axoloti to get it working again?  (if the patch had crashed, this might explain the failure on reconnection) ( did you do this after you hit connect?)</li>
<li>can you reproduce it, reliably.</li>
<li>is the CPU highly loaded?</li>
<li>when you reconnect the axoloti board, do you get a 'soft watch report' ? </li>
</ul>

<p>we have seen -7 failures before, and I've seen the verify fail too... I think when the board had crashed, <br>but I don't think I can reliably reproduce yet... it just sometimes happens, so a reproducible case would help alot.</p>
          </blockquote>
          <p><a href="https://sebiik.github.io/community.axoloti.com.backup/t/caught-exception/400/2">Read full topic</a></p>
        ]]></description>
        <link>http://community.axoloti.com/t/caught-exception/400/2</link>
        <pubDate>Fri, 04 Sep 2015 22:39:22 +0000</pubDate>
        <guid isPermaLink="false">community.axoloti.com-post-400-2</guid>
        <source url="http://community.axoloti.com/t/caught-exception/400.rss">Caught exception!</source>
      </item>
      <item>
        <title>Caught exception!</title>
        <dc:creator><![CDATA[@mackemint Markus Kaczmarek]]></dc:creator>
        <description><![CDATA[
          <p><a href="https://sebiik.github.io/community.axoloti.com.backup/u/mackemint">@mackemint</a> wrote:</p>
          <blockquote>
              <p>.. And I'm feeling proud as a cat with a mouse! <img src="/images/emoji/emoji_one/wink.png?v=0" title=":wink:" class="emoji" alt="wink"></p>

<p>This happened when triggering samples with a Korg Nanopad 2.</p>

<p>Control transfer failed: -7<br>connected<br>java.lang.Exception: cpuserial has wrong length<br>	at axoloti.HWSignature.Verify(HWSignature.java:96)<br>	at axoloti.USBBulkConnection.connect(USBBulkConnection.java:321)<br>	at axoloti.MainFrame.jCheckBoxConnectActionPerformed(MainFrame.java:769)<br>	at axoloti.MainFrame.access$400(MainFrame.java:80)<br>	at axoloti.MainFrame$7.actionPerformed(MainFrame.java:410)<br>	at javax.swing.AbstractButton.fireActionPerformed(AbstractButton.java:2022)<br>	at javax.swing.AbstractButton$Handler.actionPerformed(AbstractButton.java:2346)<br>	at javax.swing.DefaultButtonModel.fireActionPerformed(DefaultButtonModel.java:402)<br>	at javax.swing.JToggleButton$ToggleButtonModel.setPressed(JToggleButton.java:308)<br>	at javax.swing.plaf.basic.BasicButtonListener.mouseReleased(BasicButtonListener.java:252)<br>	at java.awt.Component.processMouseEvent(Component.java:6525)<br>	at javax.swing.JComponent.processMouseEvent(JComponent.java:3324)<br>	at java.awt.Component.processEvent(Component.java:6290)<br>	at java.awt.Container.processEvent(Container.java:2234)<br>	at java.awt.Component.dispatchEventImpl(Component.java:4881)<br>	at java.awt.Container.dispatchEventImpl(Container.java:2292)<br>	at java.awt.Component.dispatchEvent(Component.java:4703)<br>	at java.awt.LightweightDispatcher.retargetMouseEvent(Container.java:4898)<br>	at java.awt.LightweightDispatcher.processMouseEvent(Container.java:4533)<br>	at java.awt.LightweightDispatcher.dispatchEvent(Container.java:4462)<br>	at java.awt.Container.dispatchEventImpl(Container.java:2278)<br>	at java.awt.Window.dispatchEventImpl(Window.java:2750)<br>	at java.awt.Component.dispatchEvent(Component.java:4703)<br>	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)<br>	at java.awt.EventQueue.access$500(EventQueue.java:97)<br>	at java.awt.EventQueue$3.run(EventQueue.java:709)<br>	at java.awt.EventQueue$3.run(EventQueue.java:703)<br>	at java.security.AccessController.doPrivileged(Native Method)<br>	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)<br>	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:86)<br>	at java.awt.EventQueue$4.run(EventQueue.java:731)<br>	at java.awt.EventQueue$4.run(EventQueue.java:729)<br>	at java.security.AccessController.doPrivileged(Native Method)<br>	at java.security.ProtectionDomain$1.doIntersectionPrivilege(ProtectionDomain.java:75)<br>	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)<br>	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:201)<br>	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)<br>	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)<br>	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)<br>	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)<br>	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)</p>
          </blockquote>
          <p><a href="https://sebiik.github.io/community.axoloti.com.backup/t/caught-exception/400/1">Read full topic</a></p>
        ]]></description>
        <link>http://community.axoloti.com/t/caught-exception/400/1</link>
        <pubDate>Fri, 04 Sep 2015 22:14:44 +0000</pubDate>
        <guid isPermaLink="false">community.axoloti.com-post-400-1</guid>
        <source url="http://community.axoloti.com/t/caught-exception/400.rss">Caught exception!</source>
      </item>
  </channel>
</rss>
